resources/views/external-link-confirm.blade.php

```diff
@@ -6,61 +6,86 @@
-        title-class="header-2"
-        buttons-style="flex justify-end space-x-3"
-        init
-    >
-        @slot('title')
-            @lang('generic.external_link')
-        @endslot
+    @push('footer')
+        <x-ark-js-modal
+            :name="$identifier"
+            class="w-full max-w-2xl text-left rounded-xl"
+            title-class="header-2"
+            buttons-style="flex justify-end space-x-3"
+            x-data="{
+                hasConfirmedLinkWarning: false,
+                toggle () {
+                    this.hasConfirmedLinkWarning = ! this.hasConfirmedLinkWarning;
+                }
+            }"
+            init
+        >
+            @slot('title')
+                @lang('generic.external_link')
+            @endslot
+
+            @slot('description')
+                <div class="flex flex-col mt-8 space-y-4 whitespace-normal">
+                    <div class="font-semibold text-theme-secondary-900">
+                        <x-ark-alert type="warning" :message="$url" />
+                    </div>
+
+                    <p>@lang('generic.external_link_disclaimer')</p>
 
-        @slot('description')
-            <div class="flex flex-col mt-8 space-y-4 whitespace-normal">
-                <div class="font-semibold text-theme-secondary-900">
-                    <x-ark-alert type="warning" :message="$url" />
+                    <x-ark-checkbox
+                        name="confirmation"
+                        alpine="toggle"
+                    >
+                        @slot('label')
+                            @lang('ui::forms.do_not_show_message_again')
+                        @endslot
+                    </x-ark-checkbox>
                 </div>
-                <p>@lang('generic.external_link_disclaimer')</p>
-            </div>
-        @endslot
+            @endslot
 
-        @slot('buttons')
-            <button
-                class="button-secondary"
-                @click="hide"
-            >
-                @lang('actions.back')
-            </button>
+            @slot('buttons')
+                <button
+                    class="button-secondary"
+                    @click="hide"
+                >
+                    @lang('actions.back')
+                </button>
 
-            <a
-                target="_blank"
-                rel="noopener nofollow"
-                class="cursor-pointer button-primary"
-                href="{{ $url }}"
-                @click="hide"
-            >
-                @lang('actions.follow_link')
-            </a>
-        @endslot
-    </x-ark-js-modal>
-@endpush
+                <a
+                    target="_blank"
+                    rel="noopener nofollow"
+                    class="cursor-pointer button-primary"
+                    href="{{ $url }}"
+                    @click="hide(); hasConfirmedLinkWarning ? localStorage.setItem('has_disabled_link_warning', true) : localStorage.setItem('has_disabled_link_warning', false)"
```

Comment:
no need to store `false`, only store it when it's true and then if it doesn't exist it's considered false